### PR TITLE
Add Preporcessing Scripts and a README to the sponges tools

### PR DESCRIPTION
**Pull request by theresa-morrison**
This PR includes three pre-processing scripts I wrote and the filling script from Andrew. Together, these scripts manipulate the GLORYS data on uda into a subset region that is consistent with the input expected by `write_nudging_data.py`.

The README lists the steps for submitting these scripts. It would be gret if they could be merged and generalized to work for multiple domains.

The README also includes the overrides for MOM6 and lines that should be added to an xml to use the sponge data that is produced by these tools.

### CODE REVIEW

File: tools/sponge/README.md
Comment (uwagura) [inline tools/sponge/README.md:33]: change to "every month in a given year has been created"

File: tools/sponge/preproc_scripts/fill_glorys_nn_monthly.sh
Comment (uwagura) [inline tools/sponge/preproc_scripts/fill_glorys_nn_monthly.sh:1]: very minor, but if these are cshell scripts then maybe we should change the file extension to .csh

Comment (yichengt900) [reply tools/sponge/preproc_scripts/fill_glorys_nn_monthly.sh:1]: @theresa-morrison , since you're using C shell syntax (e.g., set), it might be simpler to rename the script to `*.csh` to reflect that.

Comment (theresa-morrison) [reply tools/sponge/preproc_scripts/fill_glorys_nn_monthly.sh]: done!

File: tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh
Comment (andrew-c-ross) [inline tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh:20]: These two dmgets could be combined into one (at least, I've always assumed dmget is happier getting everything at once instead of in multiple commands).

File: tools/sponge/preproc_scripts/get_so_monthly.csh
Comment (andrew-c-ross) [inline tools/sponge/preproc_scripts/get_so_monthly.csh:23]: Ideally these temporary files would be written to $TMPDIR and only the final result would be copied to /archive

File: tools/sponge/preproc_scripts/get_so_monthly.csh
Comment (andrew-c-ross) [inline tools/sponge/preproc_scripts/get_so_monthly.csh:29]: I know this is doing the time average, but for clarity can you add a comment stating that, and also describing what `--cnk_plc=r1d --cnk_dmn=time,1` is doing?

**Review by yichengt900 — COMMENTED** [review]
@theresa-morrison, thank you for uploading the example scripts for generating the necessary files for our sponge tools. I have tested them and can confirm that they work as expected. I'll be able to approve this PR once most of @uwagura and @andrew-c-ross's comments have been addressed. Please feel free to reach out if you need any assistance with those comments.

File: tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh
Comment (theresa-morrison) [inline tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh:22]: @andrew-c-ross  I think this combines the dmget. Nothing is being dmget in my testing, so I'm not sure if it is working.

File: tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh
Comment (theresa-morrison) [reply tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh]: changed.

File: tools/sponge/preproc_scripts/get_so_monthly.csh
Comment (theresa-morrison) [reply tools/sponge/preproc_scripts/get_so_monthly.csh]: This was meant to help the averaging be faster, but it's not neede any more.

File: tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh
Comment (theresa-morrison) [reply tools/sponge/preproc_scripts/fill_glorys_nn_monthly.csh:22]: update: not working, it is writing over my files as expected.

File: tools/sponge/preproc_scripts/get_so_monthly.csh
Comment (theresa-morrison) [reply tools/sponge/preproc_scripts/get_so_monthly.csh]: Done - only the final file is written to archive.

File: tools/sponge/preproc_scripts/merge_so_thetao_year.csh
Comment (yichengt900) [inline tools/sponge/preproc_scripts/merge_so_thetao_year.csh:25]: @theresa-morrison, sorry for being picky, but do you think it’s a good idea to reduce redundant file copies and minimize repetitive processing? We could try something like the following:
```
!/bin/tcsh
#SBATCH --ntasks=1
#SBATCH --job-name=fill_glorys_arctic
#SBATCH --time=2880
#SBATCH --partition=batch

# Usage: sbatch merge_so_thetao_year.csh <YEAR>

module load cdo
module load nco/5.0.1
module load gcp

set year = $1
set wpath = '/work/Theresa.Morrison/datasets/glorys/GLOBAL_MULTIYEAR_PHY_001_030/monthly/filled'

# Define the file variables for salinity and temperature
set so_file = "${wpath}/GLORYS_so_arctic_${year}.nc"
set thetao_file = "${wpath}/GLORYS_thetao_arctic_${year}.nc"
set final_file = "${wpath}/GLORYS_arctic_${year}.nc"

# Concatenate monthly averages into single files for salinity and temperature
foreach var (so thetao)
    ncrcat -O ${wpath}/GLORYS_${var}_arctic_${year}_*.nc ${wpath}/GLORYS_${var}_arctic_${year}.nc
end

# Append temperature data to salinity file directly without copying
ncks -A ${thetao_file} ${so_file}

# Rename the combined file to final name
mv -f ${so_file} ${final_file}
```

Comment (theresa-morrison) [reply tools/sponge/preproc_scripts/merge_so_thetao_year.csh]: I don't mind, I appreciate the suggestions!

File: tools/sponge/README.md
Comment (yichengt900) [reply tools/sponge/README.md:33]: @theresa-morrison, I think this PR is almost ready, except for a small comment from @uwagura that hasn’t been addressed yet.

Comment (theresa-morrison) [reply tools/sponge/README.md:33]: yeah, I was waiting to see if more typos would be found before submitting a commit. I will take care of this!

File: tools/sponge/README.md
Comment (yichengt900) [inline tools/sponge/README.md:42]: OOPS,  It's my bad but I found another one: "recommend"......

**Comment by theresa-morrison** [comment]
There are a few other changes I would like to make before this is merged. 
(1) I think that once the merged file is created the individual monthly filled files should be removed
(2) make domain name a variable so that it can be changed in just one place 

There is more that could be improved to streamline these scripts, but those are the two that I think make sense before merging.

**Comment by yichengt900** [comment]
Hi @theresa-morrison, I understand you have other tasks on your plate, and this one is of secondary importance. Just wanted to check in if you're still planning to make the changes you mentioned earlier, or if you'd prefer to revisit them later. Let me know what works best for you. Thanks!

**Review by yichengt900 — APPROVED** [review]
@theresa-morrison , as discussed, I’ll approve this PR for now and revisit the proposed changes at a later time. Thank you again for your efforts in enhancing the sponge tools and making them more user-friendly.

**Pull request merged**
